test/integration/targets/msc_label/tasks/main.yml

```diff
@@ -0,0 +1,283 @@
+# Test code for the MSC modules
+# Copyright: (c) 2018, Dag Wieers (@dagwieers) <dag@wieers.com>
+
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+- name: Test that we have an ACI MultiSite host, username and password
+  fail:
+    msg: 'Please define the following variables: msc_hostname, msc_username and msc_password.'
+  when: msc_hostname is not defined or msc_username is not defined or msc_password is not defined
+
+
+# CLEAN ENVIRONMENT
```

Comment:
I use blocks and failure conditions to do cleanup, is that a better approach here? Minor differences but I'd think this could muddy up a real environment.

Reply:
I always clean the environment at the start of a run (for all dependencies of the tests). But you'd normally run this on test equipment. Sure you can use blocks, but IMO they may make things more complex and may remove the environment required to troubleshoot what is going on.

So I am not a big fan of doing this myself.